### ISSUE TITLE
The application completely fails to work on Gnome (Fedora)

### ISSUE THREAD

**Issue by TheAresjej** [woelper/oculante#24]
There's no way to select a photo when you turn on the program, and the app looks like it's not supported by gnome.
![Screenshot from 2022-01-10 00-17-52](https://user-images.githubusercontent.com/19390797/148705272-e98b1dd9-4993-4c68-9a9e-55e428ae3947.png)

**Comment by woelper** [comment]
Hi @TheAresjej ! Thanks for the feedback! The application does not have a dedicated UI for opening an image. Have you tried dragging an image onto the window, as the message indicates?

Further, you should be able to "open with..." an image, selecting oculante as application. Unfortunately gnome has made that a little more complex now. Maybe this helps:
https://unix.stackexchange.com/questions/17660/default-applications-gnome-3

The application is meant to work with any operating system (and also any linux distribution, not just something with Gnome). So providing deeper Gnome support would probably be up to a packag maintainer if a linux distribution wanted to include it into it's pacakges.

Lastly, I am unsure why the window is semi-transparent - is that special to your theme/setup or does this just happen with oculante?

**Comment by TheAresjej** [comment]
_"Have you tried dragging an image onto the window, as the message indicates?"_ - **of course yes unfortunately the application does not respond.**

_"I am unsure why the window is semi-transparent"_ - **transparency is caused by bug in the application.**

_"you should be able to "open with...""_ - **I have not yet tested this option so I will see if it works and let you know.**

**Comment by woelper** [comment]
What kind of Linux distribution are you using? What desktop environment? Are you using wayland?
It might be helpful to start the application from a terminal and see if there are any error messages.

I am mainly developing this tool on Linux (PopOS & ubuntu) and have not noticed anything like that. It would be great if we could narrow down the issue more!

**Comment by TheAresjej** [comment]
_What kind of Linux distribution are you using?_ - **Fedora 34**
_What desktop environment?_ - **Gnome 40.5**
_Are you using wayland?_ - **Yes**

**Running the application via a terminal does not display any errors.**
![image](https://user-images.githubusercontent.com/19390797/149053340-e85f7473-a1b1-4804-93a2-0c087688f5f5.png)

**but running sudo solves the problem.**
![image](https://user-images.githubusercontent.com/19390797/149053487-d4d57ee5-bea1-43a4-a9b8-c2d179712961.png)

**Comment by woelper** [comment]
Thanks. It's really odd that this seems to be permission related. I found https://github.com/rust-windowing/winit/issues/306 and https://github.com/rust-windowing/winit/issues/163 which sound related. I've tested Fedora 34 in a VM and could reproduce your issue. Not too sure how to fix it though...

**Comment by woelper** [comment]
This should be fixed in the latest `0.5.21` release.

**Comment by woelper** [comment]
I'll leave a note here and a reference to #171. Please feel free to reopen this if Oculante starts failing again, but I believe we should have proper wayland support in `egui` and `notan` now to revert the env var hack. If not, I'd try something like
```rust
if std::env::var("WINIT_UNIX_BACKEND").is_err() {
    std::env::set_var("WINIT_UNIX_BACKEND", "x11");
}
```
next.
